draco/tests/test_hard.py

```diff
@@ -1,316 +1,313 @@
-from draco import is_satisfiable
-from draco.asp_utils import Block
-from draco.programs import hard, helpers
+from draco import get_violations, is_satisfiable
 
 
 def test_text_mark_without_text_channel():
-    b = hard.blocks["text_mark_without_text_channel"]
-    assert isinstance(b, Block)
-    p = helpers.program + b.program
 
-    assert is_satisfiable(
-        p
-        + """
+    assert (
+        get_violations(
+            """
     attribute((mark,type),m1,text).
     entity(encoding,m1,e1).
     attribute((encoding,channel),e1,text).
 
-    :- violation(_).
     """
+        )
+        == ["encoding_no_field_and_not_count"]
```

Comment:
Hmm, I think I would prefer not having any violations in the passing examples. What do you think?

Reply:
I was thinking that when we add more hard constraints, new violations could pop up. I was not sure whether I should try to keep some passing examples or add violations when that happens?

Reply:
Hmm good points. 

What I was thinking is that we want our tests to ensure that the constraints are written correctly, which the old format did. We should also ensure that constraints are not contained within each other because otherwise some constraints are unnecessary. However, the new format here makes it harder to test only the thing that matters. 

Let's leave this for now. I'll think more about this pull request but for now don't work on it.

Reply:
Got it. I think the old format might work better for this purpose by isolating the constraints.
But on the other hand, the new format does help with catching bugs that I wasn't aware of.

Reply:
Maybe we need separate tests. Can you send a pull request to fix the problem you found?

Reply:
Oh I mean the one above, so it's already taken care of.